tests/test_sig_stfl.c

```diff
@@ -0,0 +1,271 @@
+// SPDX-License-Identifier: MIT
+
+#if defined(_WIN32)
+#pragma warning(disable : 4244 4293)
+#endif
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include <oqs/oqs.h>
+
+#if OQS_USE_PTHREADS_IN_TESTS
+#include <pthread.h>
+#endif
+
+#ifdef OQS_ENABLE_TEST_CONSTANT_TIME
+#include <valgrind/memcheck.h>
+#define OQS_TEST_CT_CLASSIFY(addr, len)  VALGRIND_MAKE_MEM_UNDEFINED(addr, len)
+#define OQS_TEST_CT_DECLASSIFY(addr, len)  VALGRIND_MAKE_MEM_DEFINED(addr, len)
+#else
+#define OQS_TEST_CT_CLASSIFY(addr, len)
+#define OQS_TEST_CT_DECLASSIFY(addr, len)
+#endif
+
+#include "system_info.c"
+
+typedef struct magic_s {
+	uint8_t val[31];
+} magic_t;
+
+
+OQS_STATUS lock_sk_key(OQS_SECRET_KEY *sk) {
+	printf("%02x", sk->secret_key[0]);
+	return 0;
+}
+
+OQS_STATUS release_sk_key(OQS_SECRET_KEY *sk) {
+	printf("%02x", sk->secret_key[0]);
+	return 0;
+}
+
+OQS_STATUS do_nothing_save(const OQS_SECRET_KEY *sk) {
+	printf("%02x", sk->secret_key[0]);
+	return 0;
+}
+
+static OQS_STATUS sig_test_correctness(const char *method_name) {
+
+	OQS_SIG_STFL *sig = NULL;
+	uint8_t *public_key = NULL;
+	OQS_SECRET_KEY *secret_key = NULL;
+	uint8_t *message = NULL;
+	size_t message_len = 100;
+	uint8_t *signature = NULL;
+	size_t signature_len;
+	OQS_STATUS rc, ret = OQS_ERROR;
+
+	//The magic numbers are random values.
+	//The length of the magic number was chosen to be 31 to break alignment
+	magic_t magic;
+	OQS_randombytes(magic.val, sizeof(magic_t));
+
+	sig = OQS_SIG_STFL_new(method_name);
+	if (sig == NULL) {
+		fprintf(stderr, "ERROR: OQS_SIG_new failed\n");
+		goto err;
+	}
+
+	printf("================================================================================\n");
+	printf("Sample computation for signature %s\n", sig->method_name);
+	printf("================================================================================\n");
+
+	public_key = malloc(sig->length_public_key + 2 * sizeof(magic_t));
+	message = malloc(message_len + 2 * sizeof(magic_t));
+	signature = malloc(sig->length_signature + 2 * sizeof(magic_t));
+	if ((public_key == NULL) || (message == NULL) || (signature == NULL)) {
+		fprintf(stderr, "ERROR: malloc failed\n");
+		goto err;
+	}
+
+	// Define the secret key object
+	secret_key = OQS_SECRET_KEY_new(method_name);
+	secret_key->lock_key = lock_sk_key;
+	secret_key->release_key = release_sk_key;
+	secret_key->oqs_save_updated_sk_key = do_nothing_save;
+	if (secret_key == NULL) {
+		fprintf(stderr, "ERROR: OQS_SECRET_KEY_new failed\n");
+		goto err;
+	}
+
+	//Set the magic numbers before
+	memcpy(public_key, magic.val, sizeof(magic_t));
+	// memcpy(secret_key, magic.val, sizeof(magic_t));
```

Comment:
I think you can remove this commented-out line (and similar ones below) but add a comment why you're not using magic numbers for the secret key.